shared-components/statviz/Components/Visualizations/CreatedBoxes/CreatedBoxesFilterContainer.tsx

```diff
@@ -71,15 +106,17 @@ export default function CreatedBoxesFilterContainer({
       <Wrap borderWidth="1px" borderRadius="12px" padding="5" marginBottom="30px">
         <WrapItem>
           <Box width="250px">
-            <FormLabel htmlFor="box-item-select">Display by</FormLabel>
-            <Select
-              onChange={onBoxesItemsSelectChange}
-              name="box-item-select"
-              defaultValue={selectedBoxesOrItems}
-            >
-              <option value="boxesCount">Boxes</option>
-              <option value="itemsCount">Items</option>
-            </Select>
+            <SelectField
```

Comment:
If I understand you correctly you only use react-hook-form to set the default values and do validation. You are at the same time bypassing the standard onChange behavior. 
Maybe you checkout the handleSubmit return type from useForm and the mode prop of useForm. Then your onChangeProp function could be triggered by react-hook-form and it does not feel this hacky.

Reply:
ok, i will check it out but do you know how you can submit the form on update? I don't really want a submit-button or something like this.

Reply:
check this discussion out
https://github.com/orgs/react-hook-form/discussions/8486